src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantAdvertService.java

```diff
@@ -277,15 +262,15 @@ else if (section.getPages().size() == definitionSection.getPages().size()
     public void deleteGrantAdvert(UUID grantAdvertId) {
         AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
 
-        Long deletedCount = grantAdvertRepository.deleteByIdAndCreatedById(grantAdvertId, session.getGrantAdminId());
+        int deletedCount = grantAdvertRepository.deleteByIdAndSchemeEditor(grantAdvertId, session.getGrantAdminId());
 
         if (deletedCount == 0)
```

Comment:
what happens if the deleted count goes above 2147483647?  :eyes:

Reply:
I think we have bigger problems if an admin has more than 2147483647 adverts haha

So the reason i have to use int is because I have added @Modifying to the query. As a result using Long gives us this error:

Resolved [org.springframework.dao.InvalidDataAccessApiUsageException: Modifying queries can only use void or int/Integer as return type!

https://courses.baeldung.com/courses/1295711/lectures/30603968 (found this to learn about it)